dart-common/src/main/java/com/f2prateek/dart/common/AbstractDartProcessor.java

```diff
@@ -198,34 +270,41 @@ protected void parseInjectExtraAnnotatedElements(RoundEnvironment env,
     }
   }
 
-  private void parseInjectExtra(Element element, Map<TypeElement, InjectionTarget> targetClassMap,
-      Set<TypeMirror> erasedTargetTypes) {
-    TypeElement enclosingElement = (TypeElement) element.getEnclosingElement();
+  private void parseInjectExtra(Element fieldElement, Map<TypeElement, InjectionTarget>
+          targetClassMap, Set<TypeMirror> erasedTargetTypes, boolean shouldAddParentInjectionMaps) {
+    TypeElement typeElement = (TypeElement) fieldElement.getEnclosingElement();
 
     // Verify common generated code restrictions.
-    if (!isValidUsageOfInjectExtra(InjectExtra.class, element)) {
+    if (!isValidUsageOfInjectExtra(InjectExtra.class, fieldElement)) {
       return;
     }
 
-    String annotationValue = element.getAnnotation(InjectExtra.class).value();
+    InjectionTarget injectionTarget = getOrCreateTargetClass(targetClassMap, typeElement);
+    if (shouldAddParentInjectionMaps) {
+      findAllParentsInjectionMap(typeElement, injectionTarget);
```

Comment:
This is being called for every field/target, even for the ones whose ancestors are inside the same module. And for those ones we are already inhering inside `HensonExtraProcessor.process` -> enhanceInjectionTargetWithInheritedInjectionExtras

Reply:
indeed is this the best place to call this? `parseInjectExtra` is called for every field in every target, even though we make a null check inside, it gives the feeling that we are doing it multiple time (although we dont).